base/compiler/abstractinterpretation.jl

```diff
@@ -161,18 +170,22 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
             add_backedge!(edge::MethodInstance, sv)
         end
         fullmatch = false
-        for i in napplicable:-1:1
-            match = applicable[i]::SimpleVector
-            method = match[3]::Method
-            if atype <: method.sig
-                fullmatch = true
-                break
+        if length(mts) == 1
```

Comment:
When more than one MethodTable matches, it seems to me `fullmatch` should be set per MethodTable, based on the signatures that matched in that table.